documentation/src/main/asciidoc/mapper-orm-indexing-explicit.asciidoc

```diff
@@ -150,36 +110,42 @@ See <<mapper-orm-indexing-explicit-writeplan-writes>>.
 The entities will be flushed and indexed automatically.
 ====
 
-Finally, if the other solutions are unacceptable because
-the index needs to stay in its original state when the batch process fails,
-or because the process cannot be broken down in small enough transactions,
-there is a third solution: just tell Hibernate Search to process the entities
-after the call to `session.flush()` and before the call to `session.clear()`,
-without waiting for the database transaction to be committed.
-Hibernate Search will read the entities,
-build the corresponding documents and store them in an internal buffer,
-which will be written to indexes upon commit.
-
-This is done by calling the `process()` method on the write plan,
+[NOTE]
+====
+The multi-transaction solution
+and the original `flush()`/`clear()` loop pattern can be combined,
+breaking down the process in multiple medium-sized transactions,
+and periodically calling `flush`/`clear` inside each transaction.
+
+This combined solution is the most flexible,
+hence the most suitable if you want to fine-tune your batch process.
+====
+
+If breaking down the batch process into multiple transactions is not an option,
+a second solution is to just write to indexes
+after the call to `session.flush()`/`session.clear()`,
+without waiting for the database transaction to be committed:
+the internal document buffer will be cleared after each write to indexes.
+
+This is done by calling the `execute()` method on the write plan,
 as shown in the example below.
 
 [IMPORTANT]
 ====
-This pattern will require less memory than without the periodic `flush`/`process`/`clear`,
-but it will still maintain an internal buffer that will grow as more entities are processed,
-and may lead to an `OutOfMemoryException` depending on the JVM heap size and the complexity and number of documents.
-Only use this in production after careful testing with an appropriate JVM heap size.
-
-However, if enough memory is available,
-this is the safest pattern, leaving both the database and indexes unaffected
-if an error happens during the batch process.
+With this pattern, if an exception is thrown,
+part of the data will already be in the index, with no way to roll back the changes,
+while the database changes will have been rolled back.
+The index will thus be inconsistent with the database.
+
+To recover from that situation, you will have to either cancel your database changes manually,
+or wipe the indexes clean and <<mapper-orm-indexing-explicit-massindexer,reindex>>.
```

Comment:
Maybe it is too scary. Usually the user can just resume the batch process from the first not committed chunk. I would mention this option too.

Reply:
In a recovery of consistency process a forward compensation can be easier and safer than a backward one.

Reply:
This is scary on purpose, because this strategy is a bad idea. But I'm sure some people will want it anyway, so I mention it for completeness.

That being said, you're right that there's something wrong here: there are no database changes to cancel, since they were rolled back.
I'll change this part.